src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs

```diff
@@ -62,20 +61,47 @@ public async Task<IActionResult> Authorize()
             return NotFound();
         }
 
-        // Retrieve the claims stored in the authentication cookie.
-        // If they can't be extracted, redirect the user to the login page.
+        // Try to retrieve the user principal stored in the authentication cookie and redirect
+        // the user agent to the login page (or to an external provider) in the following cases:
+        //
+        //  - If the user principal can't be extracted or the cookie is too old.
+        //  - If prompt=login was specified by the client application.
+        //  - If max_age=0 was specified by the client application (max_age=0 is equivalent to prompt=login).
+        //  - If a max_age parameter was provided and the authentication cookie is not considered "fresh" enough.
         var result = await HttpContext.AuthenticateAsync();
-        if (result == null || !result.Succeeded || request.HasPromptValue(PromptValues.Login))
+        if (result is not { Succeeded: true } ||
+            ((request.HasPromptValue(PromptValues.Login) || request.MaxAge is 0 ||
+             (request.MaxAge is not null && result.Properties?.IssuedUtc is not null &&
```

Comment:
For some reason, the standard `max_age` parameter wasn't supported: this PR fixes that by implementing the expected behavior.